judger/src/tester/exec.rs

```diff
@@ -90,27 +103,46 @@ impl Step {
     }
 
     /// Make a new `Step` with a `timeout`.
-    pub fn new_with_timeout(cmd: Capturable, timeout: Option<time::Duration>) -> Self {
-        Step { cmd, timeout }
+    pub fn new_with_timeout(
+        cmd: Capturable,
+        timeout: Option<time::Duration>,
+        is_user_command: bool,
+    ) -> Self {
+        Step {
+            cmd,
+            is_user_command,
+            timeout,
+        }
     }
 
     /// Run the `Step` and collect its info, considering the `timeout`.
-    pub async fn capture<R>(self, runner: &R) -> PopenResult<ProcessInfo>
+    pub async fn capture<R>(
+        self,
+        runner: &R,
+        variables: &HashMap<String, String>,
+    ) -> PopenResult<ProcessInfo>
     where
         R: CommandRunner + Send,
     {
```

Comment:
Try early return style here?

Reply:
Doesn't seem to simplify much... Replaced with `map_err` and `map` instead.